cpp/src/arrow/compute/kernels/scalar_boolean.cc

```diff
@@ -467,10 +467,10 @@ const FunctionDoc or_kleene_doc{
     "Logical 'or' boolean values (Kleene logic)",
     ("This function behaves as follows with nulls:\n\n"
      "- true or null = true\n"
-     "- null and true = true\n"
-     "- false and null = null\n"
-     "- null and false = null\n"
-     "- null and null = null\n"
+     "- null or true = true\n"
+     "- false or null = null\n"
+     "- null or false = null\n"
+     "- null or null = null\n"
```

Comment:
Thanks for the fix. Looks line 477 also needs update.
`see function \"and\"`  --> `see function \"or\"`

Reply:
thanks, I updated that, too